optics-core/src/Optics/Internal/Optic.hs

```diff
@@ -115,7 +114,7 @@ castOptic (Optic o) = Optic (cast o)
 -- indexed, the composition preserves all the indices.
 --
 infixl 9 %
-(%) :: (Is k m, Is l m, m ~ Join k l, ks ~ Append is js)
+(%) :: (Is k m, Is l m, m ~ Join k l, AppendIndices is js, ks ~ Append is js)
```

Comment:
I think this is generally a good idea, but the only thing that is a slight shame is making the type of `%` more complicated, with `AppendIndices` potentially showing up in inferred types. Perhaps it is worth making `AppendIndices` takes `ks` as a parameter so that `AppendIndices is js ks` can imply `ks ~ Append is js`?

Reply:
One could maybe get rid of `Append` family all together, by having 

```diff
-appendIndices :: IxEq i (Curry xs (Curry ys i)) (Curry (Append xs ys) i)
+appendIndices :: IxEq i (Curry xs (Curry ys i)) (Curry zs i)
```

... but I guess we cannot as there is no sane way to break functional dependencies in GHCs.

Reply:
> Perhaps it is worth making AppendIndices takes ks as a parameter

That's what I've done in #353, but then you pointed out [here](https://github.com/well-typed/optics/pull/353#issuecomment-687915488) that GHC expands `ks` to be `Append is js` and removing it would be better ;)

Reply:
> One could maybe get rid of Append family all together

I tried that and looks like it works, so I re-added the third parameter and removed dependence on `Append`.